torch/fx/experimental/proxy_tensor.py

```diff
@@ -757,9 +759,101 @@ def disable_autocast_cache():
+    being the actual module being traced. Since make_fx traces a lambda of our
+    creation, things don't work properly.
+
+    So for this version we hold onto a reference to the original module
+    (scope_root) and monkeypatch it in when capturing the stack trace.
+
+    TODO: we should probably augment the standard FX APIs to better support how
+    make_fx uses it.
+    """
+
+    def __init__(self, scope_root):
+        super().__init__()
+        self.scope_root = scope_root
+        self.proxy_paths = WeakKeyDictionary()
+        self.proxy_modules = WeakKeyDictionary()
+
+        self.module_id_cache = defaultdict(list)
+        for name, mod in self.scope_root.named_modules(remove_duplicate=False):
+            self.module_id_cache[id(mod)].append(name)
+
+        self_ = self
+
+        class AttrProxy:
+            def __init__(self, base, path):
+                self.__class__ = type(
+                    base.__class__.__name__,
+                    (self.__class__, base.__class__),
+                    {},
+                )
+                self.__dict__ = base.__dict__
+                self_.proxy_paths[self] = path
+                self_.proxy_modules[self] = base
+
+            def __getattr__(self, name):
+                assert isinstance(self, torch.nn.Module)
+                attr_val = super().__getattr__(name)
+                if isinstance(attr_val, AttrProxy):
+                    attr_val = self_.proxy_modules[attr_val]
+                elif not isinstance(attr_val, torch.nn.Module):
+                    return attr_val
+                return AttrProxy(attr_val, self_.proxy_paths[self] + "." + name)
+
+            @property
+            def _modules(self):
+                assert "_modules" in self.__dict__
+                submodules = self.__dict__["_modules"]
+                assert isinstance(submodules, dict)
+                return {
+                    key: AttrProxy(value, self_.proxy_paths[self] + "." + str(key))
+                    for key, value in submodules.items()
+                }
+
+        self.proxy_type = AttrProxy
+
+    def path_of_module(self, mod: torch.nn.Module) -> str:
+        """
+        Use tracked access path during tracing instead of the default BFS behavior.
+        Still use all the possible module paths to verify the result.
+        """
+        if mod is self.scope_root:
+            return ""
+        assert isinstance(mod, self.proxy_type)
+        module_id = id(self.proxy_modules[mod])
+        results = self.module_id_cache[module_id]
+        assert self.proxy_paths[mod] in results
+
+        return self.proxy_paths[mod]
+
+    def getattr(self, attr, attr_val, parameter_proxy_cache):
+        if not isinstance(attr_val, torch.nn.Module):
+            return super().getattr(attr, attr_val, parameter_proxy_cache)
+        return self.proxy_type(attr_val, attr)
+
+    def call_module(self, m, forward, args, kwargs):
```

Comment:
This excludes the `preserve_signature` handling of the version in aot_export. We need this logic, right? Otherwise `preserve_signature` does not function correctly for non-strict mode?

I'm okay with taking this as a follow-up, but let's make sure to track it

Reply:
Yep will do that as follow up.